tests/tecc

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Simo Sorce <simo@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source ${TESTSSRCDIR}/helpers.sh
+
+title PARA "Export EC Public key to a file"
+ossl 'pkey -in $ECPUBURI -pubin -pubout -out ${TMPPDIR}/ecout.pub'
+title LINE "Print EC Public key from private"
+ossl 'pkey -in $ECPRIURI -pubout -text' output
+FAIL=0
+echo $output | grep "PKCS11 EC Public Key (256 bits)" > /dev/null 2>&1 || FAIL=1
+if [ $FAIL -eq 1 ]; then
+    echo "Pkcs11 encoder function failed"
+    echo
+    echo "Original command output:"
+    echo "$output"
+    echo
+fi
+
+title PARA "Sign and Verify with provided Hash and EC"
+ossl '
+pkeyutl -sign -inkey "${ECBASEURI}"
+              -in ${TMPPDIR}/sha256.bin
+              -out ${TMPPDIR}/sha256-ecsig.bin'
+
+ossl '
+pkeyutl -verify -inkey "${ECBASEURI}" -pubin
+                -in ${TMPPDIR}/sha256.bin
+                -sigfile ${TMPPDIR}/sha256-ecsig.bin'
+
+ossl '
+pkeyutl -verify -inkey "${TMPPDIR}/ecout.pub" -pubin
+                -in ${TMPPDIR}/sha256.bin
+                -sigfile ${TMPPDIR}/sha256-ecsig.bin'
+
+title PARA "DigestSign and DigestVerify with ECC"
+ossl '
+pkeyutl -sign -inkey "${ECBASEURI}"
+              -digest sha256
+              -in ${RAND64FILE}
+              -rawin
+              -out ${TMPPDIR}/sha256-ecdgstsig.bin'
+ossl '
+pkeyutl -verify -inkey "${ECBASEURI}" -pubin
+                -digest sha256
+                -in ${RAND64FILE}
+                -rawin
+                -sigfile ${TMPPDIR}/sha256-ecdgstsig.bin'
```

Comment:
would it make sense to give a shot to some other digests too, just to get more test coverage?

Reply:
yes, I think that is a reasonable thing to do.

Reply:
added sha384 and sha512, I do not think I can reasonably add more until we have a better way to exclude specific tests.
NSS softokn does not support SHA3 for example.

Reply:
I guess if I make additional Digest fallback to OpenSSL digest I might be able to also to SHA3 ... Should we do that as part of this PR ?
It is possible that a token that does not support DigestSign APIs does not support Digest APIs at all after all ...

Reply:
Sounds like separate PR. SoftHSM will not support SHA3 probably either as far as I know. But its usage in the applications is not very wide so I think it should not be an issue.
In the next PR, I think this can be extended to allow fallback to OpenSSL implementation of digests or disabling this functionality through configuration file if this might be a concern to a certification.